asapdiscovery-simulation/asapdiscovery/simulation/schema.py

```diff
@@ -0,0 +1,33 @@
+from collections import namedtuple
+
+from openmm.app import PME, HBonds
+from openmm.unit import amu, nanometers
+
+ForceFieldParams = namedtuple(
```

Comment:
I would just make this a schema as more flexible.

Reply:
I don't know how to do this, as validating the various `PME`, `nanometers`, etc is not obvious to me.

Reply:
I agree though! I just went with the named tuple because it gave me the nice features of having some expected fields without having to validate the objects themselves

Reply:
I wouldn't worry about validation, a `Field` would be enough at this stage, but we can push this down the line if you want.

Reply:
What I get with the current version:

```
pydantic.error_wrappers.ValidationError: 5 validation errors for ForceFieldParams
padding
  instance of UndefinedType expected (type=type_error.arbitrary_type; expected_arbitrary_type=UndefinedType)
nonbonded_method
  instance of UndefinedType expected (type=type_error.arbitrary_type; expected_arbitrary_type=UndefinedType)
nonbonded_cutoff
  instance of UndefinedType expected (type=type_error.arbitrary_type; expected_arbitrary_type=UndefinedType)
constraints
  instance of UndefinedType expected (type=type_error.arbitrary_type; expected_arbitrary_type=UndefinedType)
hydrogen_mass
  instance of UndefinedType expected (type=type_error.arbitrary_type; expected_arbitrary_type=UndefinedType)
```

Reply:
something like 

```python 
from typing import List
from pydantic import BaseModel, Field

class ForceFieldParams(BaseModel):
    ff_xmls: List[str] = Field(..., description="List of FF XMLs")
    padding: float = Field(..., description="Padding value")
    model: str = Field(..., description="Model value")
    nonbondedMethod: str = Field(..., description="Nonbonded method")
    nonbondedCutoff: float = Field(..., description="Nonbonded cutoff value")
    ewaldErrorTolerance: float = Field(..., description="Ewald error tolerance")
    constraints: str = Field(..., description="Constraints value")
    rigidWater: bool = Field(..., description="Rigid water")
    hydrogenMass: float = Field(..., description="Hydrogen mass")

    def to_openmm(self):
     ...
```

will probably work, but I take your point that can't store `Hbonds` or `PME` in a schema. Perhaps easier to just use your namedtuple.

Reply:
ok, going back to named tuple here: https://github.com/choderalab/asapdiscovery/pull/238/commits/c43f23ae51f56c9b61143d41ffae39d3e6e8ab33

Reply:
we'll go with the namedtuple!